src/lib.rs

```diff
@@ -541,7 +764,7 @@ mod tests {
             _reason: &str,
         ) -> Result<(), Self::Error> {
             // Actual blocks taken from an SD card, except I've changed the start and length of partition 0.
```

Comment:
I added a fake `info_sector`, because I didn't want to change all the blocks, if you still have the SD image you can copy in the third block and run cargo test, I think it should be fine, I tested this on other fat32 images.